scale/job/execution/configuration/configurators.py

```diff
@@ -64,7 +68,29 @@ def configure_queued_job(self, job):
 
         # Set up env vars for job's input data
         input_values = data.get_injected_input_values(input_files_dict)
-        env_vars = data.get_injected_env_vars(input_files_dict)
+        interface = None
+        if JobInterfaceSunset.is_seed_dict(job.job_type.manifest):
+            interface = SeedManifest(job.job_type.manifest, do_validate=False).get_input_interface()
+        elif job.job_type.manifest and 'input_data' in job.job_type.manifest:
+            # TODO: This can be removed when support for legacy job types is removed
+            interface = Interface()
+            for input_dict in job.job_type.manifest['input_data']:
+                media_types = input_dict['media_types'] if 'media_types' in input_dict else []
+                required = input_dict['required'] if 'required' in input_dict else True
+                if input_dict['type'] == 'file':
+                    param = FileParameter(input_dict['name'], media_types, required, False)
+                    interface.add_parameter(param)
+                elif input_dict['type'] == 'files':
+                    param = FileParameter(input_dict['name'], media_types, required, True)
+                    interface.add_parameter(param)
+                elif input_dict['type'] == 'property':
+                    interface.add_parameter(JsonParameter(input_dict['name'], 'string', required))
+
+        env_vars = {}
+        if isinstance(data, JobData):
+            env_vars = data.get_injected_env_vars(input_files_dict, interface)
+        else:
+            env_vars = data.get_injected_env_vars(input_files_dict)
```

Comment:
Assuming this calls the `JobData.get_injected_env_vars`, I would expect an error to be thrown here since the method signature doesn't define a default value for `interface`.